packages/core/src/api/guild.ts

```diff
@@ -0,0 +1,713 @@
+		return (await this.rest.patch(Routes.guild(guildId), {
+			reason,
+			body: guild,
+		})) as APIGuild;
+	}
+
+	/**
+	 * Deletes a guild
+	 *
+	 * @param guildId - The id of the guild to delete
+	 * @param reason - The reason for deleting this guild
+	 */
+	public async delete(guildId: string, reason?: string) {
+		return (await this.rest.delete(Routes.guild(guildId), { reason })) as APIGuild;
+	}
+
+	/**
+	 * Fetches all the members of a guild
+	 *
+	 * @param guildId - The id of the guild
+	 * @param options - The options to use when fetching the guild members
+	 */
+	public async getAll(guildId: string, options: RESTGetAPIGuildMembersQuery = {}) {
+		return (await this.rest.get(Routes.guildMembers(guildId), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+		})) as APIGuildMember[];
+	}
+
+	/**
+	 * Fetches a guild's channels
+	 *
+	 * @param guildId - The id of the guild to fetch the channels for
+	 */
+	public async getChannels(guildId: string) {
+		return (await this.rest.get(Routes.guildChannels(guildId))) as APIChannel[];
+	}
+
+	/**
+	 * Creates a guild channel
+	 *
+	 * @param guildId - The id of the guild to create the channel in
+	 * @param channel - The data to create the new channel
+	 * @param reason - The reason for creating this channel
+	 */
+	public async createChannel(guildId: string, channel: RESTPostAPIGuildChannelJSONBody, reason?: string) {
+		return (await this.rest.post(Routes.guildChannels(guildId), {
+			reason,
+			body: channel,
+		})) as APIChannel;
+	}
+
+	/**
+	 * Edits a guild channel's positions
+	 *
+	 * @param guildId - The id of the guild to edit the channel positions for
+	 * @param options - The options to edit the channel positions with
+	 * @param reason - The reason for editing the channel positions
+	 */
+	public async setChannelPosition(
+		guildId: string,
+		options: RESTPatchAPIGuildChannelPositionsJSONBody,
+		reason?: string,
+	) {
+		return this.rest.patch(Routes.guildChannels(guildId), {
+			reason,
+			body: options,
+		});
+	}
+
+	/**
+	 * Fetches the active threads in a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the active threads for
+	 */
+	public async getActiveThreads(guildId: string) {
+		return (await this.rest.get(Routes.guildActiveThreads(guildId))) as APIThreadChannel[];
+	}
+
+	/**
+	 * Fetches a guild member ban
+	 *
+	 * @param guildId - The id of the guild to fetch the ban for
+	 */
+	public async getBans(guildId: string) {
+		return (await this.rest.get(Routes.guildBans(guildId))) as APIBan[];
+	}
+
+	/**
+	 * Bans a guild member
+	 *public
+	 *
+	 * @param guildId - The id of the guild to ban the member in
+	 * @param userId - The id of the user to ban
+	 * @param options - Options for banning the user
+	 * @param reason - The reason for banning the user
+	 */
+	public async ban(guildId: string, userId: string, options: RESTPutAPIGuildBanJSONBody = {}, reason?: string) {
```

Comment:
```suggestion
	public async banUser(guildId: string, userId: string, options: RESTPutAPIGuildBanJSONBody = {}, reason?: string) {
```
We don't ban a guild